cranelift/filetests/filetests/runtests/call.clif

```diff
@@ -0,0 +1,67 @@
+test run
+target x86_64
+target aarch64
+target aarch64 sign_return_address
```

Comment:
@akirilov-arm Do we need more variations of this?

Right now this runs the tests with both `sign_return_address` on and off, but with `has_pauth` always disabled (even if the host supports it). 

My questions here are more directed towards, do we need `sign_return_address_all`? And if we have `has_pauth` enabled and we start validating addresses, can we trust that the A key is the correct one to use (unless specified otherwise like it is for MacOSX)?

Reply:
While it is always better to check more options, `sign_return_address` covers probably at least 90% of the problem space. As for `has_pauth`, I am not sure what you mean - currently that setting changes just one small code generation detail (whether to use `RETAA`/`RETAB` instead of `AUTIASP`/`AUTIBSP` + `RET`), and only when used in combination with `sign_return_address` (it has no effect otherwise). Concerning the A key, Linux does not impose any particular key choice, even within the same process - the only important detail is to be consistent, that is sign and authenticate a specific pointer value with the same key.

Reply:
That's good to know! I didn't know if adding `has_pauth` would affect these test cases in any meaningful way. But we might as well test those other cases with `has_pauth`.

I guess we can leave the more advanced cases for later.

Reply:
I am sorry, but I forgot to mention one important detail - the `RETAA`/`RETAB` instruction won't be recognized by hardware that does not support pointer authentication, so it will cause a processor exception, and that's why we can't just enable `has_pauth` here. Given that `sign_return_address` is also enabled, `has_pauth` basically implies generating code that is **not** backward compatible, , i.e. non-`HINT` instructions, so it is similar to, say, `has_avx512vl`.

Note that this is **not** an issue with `sign_return_address` by itself.

Reply:
That should be ok, we only run these tests if `cranelift-native` detects that the host `has_pauth`, otherwise they get skipped in the runtest suite. It won't run the test on every machine, but at least qemu seem to emulate pauth and should run all of these (I haven't confirmed this though).

Reply:
Ah, I didn't realize that the feature detection logic in `cranelift-native` overrode the directives in the test files. In that case there is no issue, indeed, and you are absolutely correct that QEMU supports pretty much all ISA extensions that we use.